supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/authentication/PipelinesApiVMAuthentication.scala

```diff
@@ -34,19 +35,24 @@ object PipelinesApiDockerCredentials {
 
   def apply(dockerCredentials: DockerCredentials, googleConfig: GoogleConfiguration): PipelinesApiDockerCredentials = {
     // If there's an encryption key defined there must be a valid auth defined to encrypt it.
-    val credential = dockerCredentials.keyName flatMap { _ =>
-      val authName = dockerCredentials.authName.getOrElse(throw new RuntimeException("KMS Encryption key defined for private Docker but no auth specified"))
-      googleConfig.auth(authName) match {
-        case Invalid(e) => throw new RuntimeException("Error creating GoogleAuthMode: " + e.toList.mkString(", "))
-        case Valid(auth) => auth.apiClientGoogleCredential(identity)
-      }
+    val authValidation = dockerCredentials.keyName match {
+      case None => ().validNel // fine
+      case _ =>
+        for {
+          authName <- dockerCredentials.authName.toErrorOr("KMS Encryption key defined for private Docker but no auth specified")
+          _ <- googleConfig.auth(authName)
+        } yield ()
     }
 
-    new PipelinesApiDockerCredentials(
-      token = dockerCredentials.token,
-      keyName = dockerCredentials.keyName,
-      authName = dockerCredentials.authName,
-      credential = credential)
+    authValidation match {
+      case Invalid(errors) =>
+        throw new RuntimeException(errors.toList.mkString(", "))
+      case Valid(_) =>
+        new PipelinesApiDockerCredentials(
```

Comment:
If we return Checked we can move this code up to L39 and skip this second pattern match